debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/ChainedReader.java

```diff
@@ -22,48 +23,61 @@
  * This reader ensures that all records generated by one of its contained {@link Reader}s are all passed through to callers
  * via {@link #poll() polling} before the next reader is started. And, when this reader is {@link #stop() stopped}, this
  * class ensures that current reader is stopped and that no additional readers will be started.
- * 
+ *
  * @author Randall Hauch
  */
 @ThreadSafe
 public final class ChainedReader implements Reader {
 
     private final Logger logger = LoggerFactory.getLogger(getClass());
-    private final List<Reader> readers = new ArrayList<>();
+    private final List<Reader> readers;
+    private final String completionMessage;
     private final LinkedList<Reader> remainingReaders = new LinkedList<>();
     private final AtomicBoolean running = new AtomicBoolean();
     private final AtomicBoolean completed = new AtomicBoolean(true);
     private final AtomicReference<Reader> currentReader = new AtomicReference<>();
     private final AtomicReference<Runnable> uponCompletion = new AtomicReference<>();
-    private final AtomicReference<String> completionMessage = new AtomicReference<>();
 
-    /**
-     * Create a new chained reader.
-     */
-    public ChainedReader() {
+    public static class Builder {
+
+        private final List<Reader> readers = new ArrayList<>();
+        private String completionMessage;
+
+        public Builder addReader(Reader reader) {
+            readers.add(reader);
+            return this;
+        }
+
+        /**
+         * Set the message that should be logged when all of the readers have completed their work.
+         */
+        public Builder completionMessage(String message) {
```

Comment:
Wouldn't it be better to provide a completionAction instead of completionMessage and pass a consumer to be called where the default consumer will be the loggin one?

Reply:
There already is `uponCompletion(Runnable)`, but I didn't want to touch that for now to avoid too many trouble with DBZ-175. Also this message is only set for the blocking consumer, which we can get rid of when adding the event queue. So I think it's ok for the time being.

Reply:
Makes sense, thanks